railties/lib/rails/test_unit/runner.rb

```diff
@@ -0,0 +1,140 @@
+require "shellwords"
+require "method_source"
+require "rake/file_list"
+require "active_support/core_ext/module/attribute_accessors"
+
+module Rails
+  module TestUnit
+    class Runner
+      mattr_reader :filters, default: []
+
+      class << self
+        def options(opts)
+          opts.on("--warnings", "-w", "Enable Ruby warnings") {}
+          opts.on("--environment", "-e", "Run tests in the ENV environment") {}
+        end
+
+        def parse_options(argv)
+          # Perform manual parsing and cleanup since option parser raises on unknown options.
+          env_index = argv.index("--environment") || argv.index("-e")
+          if env_index
+            argv.delete_at(env_index)
+            environment = argv.delete_at(env_index).strip
+          end
+          ENV["RAILS_ENV"] = environment || "test"
+
+          w_index = argv.index("--warnings") || argv.index("-w")
+          $VERBOSE = argv.delete_at(w_index) if w_index
```

Comment:
I really despise this code. At first I had defined these as Thor options in the test command, but I couldn't figure out a way to make Thor remove the options from ARGV after parsing them (that also broke the test/generators/plugin_test_runner_test.rb tests).